File: Leetcode/1072. Flip Columns For Maximum Number of Equal Rows.java
class Solution {
    public int maxEqualRowsAfterFlips(int[][] matrix) {
        Map<String, Integer> map = new HashMap<>();
        int ans = 0;
        for (int[] row : matrix) {
            StringBuilder sb = new StringBuilder();
            for (int cell : row) sb.append(cell == row[0] ? 'T' : 'F');
            ans = Math.max(ans, map.merge(sb.toString(), 1, Integer::sum));
        }
        return ans;
    }
}

File: C#/1072. Flip Columns For Maximum Number of Equal Rows.cs
public class Solution {
    public int MaxEqualRowsAfterFlips(int[][] matrix) {
        var map = new Dictionary<string, int>();
        int ans = 0;
        foreach (var row in matrix) {
            var key = string.Join("", row.Select(cell => cell == row[0] ? 'T' : 'F'));
            map[key] = map.GetValueOrDefault(key, 0) + 1;
            ans = Math.Max(ans, map[key]);
        }
        return ans;
    }
}
